spark/src/main/java/org/apache/iceberg/spark/SparkExceptionUtil.java

```diff
@@ -39,7 +40,9 @@ private SparkExceptionUtil() {
    * @param args format specifiers
    * @return unchecked exception.
    */
-  public static RuntimeException toUncheckedException(Throwable cause, String message, Object... args) {
+  @FormatMethod
+  public static RuntimeException toUncheckedException(final Throwable cause, final String message,
```

Comment:
FormatMethod needs all variables to be effectively final. We do a lot of casting here which maybe screws up error-prone's analysis because we don't actually change anything here but ErrorProne still complains. Declaring everything final fixes it.

Reply:
You'll find no complaints from me for marking variables as `final`.

Reply:
Can we format it like this?

```
public static RuntimeException toUncheckedException(final Throwable cause, final String message,
                                                    final Object... args)
```

or


```
public static RuntimeException toUncheckedException(
    final Throwable cause, final String message, final Object... args)
```

Can we also add a comment clarifying why we need final args? We don't usually do that in the rest of the code.